gemd/entity/object/base_object.py

```diff
@@ -52,19 +60,52 @@ def _attribute_has_setter(cls, name: str) -> bool:
         return prop is None or prop.fset is not None
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Get name."""
         return self._name
 
     @name.setter
-    def name(self, name):
+    def name(self, name: str):
         self._name = validate_str(name)
 
     @property
-    def file_links(self):
+    def file_links(self) -> List[FileLink]:
         """Get file links."""
         return self._file_links
 
     @file_links.setter
-    def file_links(self, file_links):
+    def file_links(self, file_links: Union[Iterable[FileLink], FileLink]):
         self._file_links = validate_list(file_links, FileLink)
+
+    def all_dependencies(self) -> Set[BaseEntity]:
+        """Return a set of all immediate dependencies (no recursion)."""
+        from gemd.entity.object.has_parameters import HasParameters
+        from gemd.entity.object.has_conditions import HasConditions
+        from gemd.entity.object.has_properties import HasProperties
+        from gemd.entity.object.has_spec import HasSpec
+        from gemd.entity.object.has_template import HasTemplate
+
+        from gemd.entity.object import IngredientRun, MaterialRun, MeasurementRun  # no ProcessRun
+        from gemd.entity.object import IngredientSpec, MaterialSpec  # no ProcessSpec
+
+        result = set()
+
+        for typ in (HasParameters, HasConditions, HasProperties, HasSpec, HasTemplate):
+            if isinstance(self, typ):
+                result |= typ.all_dependencies(self)
```

Comment:
Why use the in-place union operation here, but `result.add` below? They seem to me like they both do the same thing.

Reply:
`add` adds a single element to the set, where as `|=` adds all elements of a second set (like `append` & `extend` for lists).  If you'd prefer all method calls, `result |=` could be changed to `result.update()`

Reply:
@bfolie For clarity, I did not change `|=` to `result.update()` in my last push.  Should I have?

Reply:
No need to change